Sources/_CryptoExtras/MLDSA/MLDSA_boring.swift

```diff
@@ -0,0 +1,319 @@
+                let result = data.regions.first!.withUnsafeBytes { dataPtr in
+                    let contextPointer = context?.withUnsafeBufferPointer { $0.baseAddress }
+                    let contextLength = context?.count ?? 0
+
+                    return CCryptoBoringSSL_MLDSA65_sign(
+                        bufferPtr.baseAddress,
+                        self.pointer,
+                        dataPtr.baseAddress,
+                        dataPtr.count,
+                        contextPointer,
+                        contextLength
+                    )
+                }
+
+                guard result == 1 else {
+                    throw CryptoMLDSAError.signatureGenerationFailure
+                }
+
+                length = Signature.bytesCount
+            }
+            return Signature(signatureBytes: output)
+        }
+
+        /// The size of the private key in bytes.
+        public static let bytesCount = 4032
+    }
+}
+
+extension MLDSA {
+    /// A ML-DSA-65 public key.
+    public struct PublicKey: Sendable {
+        private let pointer: UnsafeMutablePointer<MLDSA65_public_key>
+
+        fileprivate init(privateKey: PrivateKey) throws {
+            self.pointer = UnsafeMutablePointer<MLDSA65_public_key>.allocate(capacity: 1)
+
+            guard CCryptoBoringSSL_MLDSA65_public_from_private(
+                self.pointer,
+                privateKey.pointer
+            ) == 1 else {
+                throw CryptoMLDSAError.keyGenerationFailure
+            }
+        }
+
+        /// Initialize a ML-DSA-65 public key from a seed.
+        /// 
+        /// The seed must be at least 32 bytes long.
+        /// Any additional bytes in the seed are ignored.
+        /// 
+        /// - Parameter seed: The seed to use to generate the public key.
+        public init(from seed: some DataProtocol) throws {
+            try self.init(privateKey: PrivateKey(from: seed))
+        }
+
+        /// Initialize a ML-DSA-65 public key from a DER representation.
+        /// 
+        /// - Parameter derRepresentation: The DER representation of the public key.
+        /// 
+        /// - Throws: `CryptoKitError.incorrectKeySize` if the DER representation is not the correct size or ``CryptoMLDSAError/keyGenerationFailure`` if the key could not be generated.
+        public init(derRepresentation: some DataProtocol) throws {
+            guard derRepresentation.count == MLDSA.PublicKey.bytesCount else {
+                throw CryptoKitError.incorrectKeySize
+            }
+
+            self.pointer = UnsafeMutablePointer<MLDSA65_public_key>.allocate(capacity: 1)
+
+            let cbsPointer = UnsafeMutablePointer<CBS>.allocate(capacity: 1)
+            defer { cbsPointer.deallocate() }
+            derRepresentation.regions.flatMap { $0 }.withUnsafeBufferPointer { buffer in
+                cbsPointer.pointee = CBS(data: buffer.baseAddress, len: buffer.count)
```

Comment:
You can't do this: this propagates the pointer for the bytes out of the closure, which is forbidden. We need to use the CBS within this closure.